services/topology-engine/queue-engine/topologylistener/isl_utils.py

```diff
@@ -20,10 +20,71 @@
 
 from topologylistener import db
 from topologylistener import exc
+from topologylistener import model
 
 logger = logging.getLogger(__name__)
 
 
+def fetch(tx, isl):
+    match = _make_match(isl)
+    q = textwrap.dedent("""
+        MATCH
+          (:switch {name: $src_switch})
+          -
+          [target:isl {
+            src_switch: $src_switch,
+            src_port: $src_port,
+            dst_switch: $dst_switch,
+            dst_port: $dst_port
+          }]
+          ->
+          (:switch {name: $dst_switch})
```

Comment:
As for me this way of formatting makes Cypher query less readable. We should discuss the better way...